### PR TITLE
aseq2json: init at 2018-04-28

### PR DESCRIPTION
**Pull request by queezle42**
###### Motivation for this change
Adds a package for `aseq2json`, which is a program that connects to the ALSA sequencer and dumps midi events as json.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/aseq2json/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    owner = "google";
+    repo = "midi-dump-tools";
+    rev = "8572e6313a0d7ec95492dcab04a46c5dd30ef33a";
+    sha256 = "LQ9LLVumi3GN6c9tuMSOd1Bs2pgrwrLLQbs5XF+NZeA=";
+  };
+in
+  stdenv.mkDerivation {
+    pname = "aseq2json";
+    version = "git";
+    src = "${repository}/aseq2json";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/aseq2json/default.nix:14]:
```suggestion
stdenv.mkDerivation rec {
  pname = "aseq2json";
  version = "unstable-XXXX-XX-XX";
    
  repository = fetchFromGitHub {
    owner = "google";
    repo = "midi-dump-tools";
    rev = "8572e6313a0d7ec95492dcab04a46c5dd30ef33a";
    sha256 = "LQ9LLVumi3GN6c9tuMSOd1Bs2pgrwrLLQbs5XF+NZeA=";
  };
    
    sourceRoot = "aseq2json";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
also the indentation seems to be wrong.

**Comment by qzle** [comment]
@SuperSandro2000 It looks like part of your review got lost, but I hope everything is fixed now.

File: pkgs/os-specific/linux/aseq2json/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, lib, fetchFromGitHub, pkg-config, alsaLib, glib, json-glib }:
+
+let
+  repository = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/aseq2json/default.nix:4]: Removing that let in here and using sourceRoot makes the package more similar to other packages which saves headaches in the long run.

Comment (qzle) [reply pkgs/os-specific/linux/aseq2json/default.nix]: That makes it a lot cleaner, thanks!

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
You forgot to add the package to top-level nix.

**Comment by qzle** [comment]
Thanks for the patience, it's my first package :)
Is there anything else missing?

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -29890,4 +29890,6 @@ in
   lc3tools = callPackage ../development/tools/lc3tools {};
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  aseq2json = callPackage ../os-specific/linux/aseq2json {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:29894]: Please sort this into a fitting category and sort there alphabetically.

Comment (qzle) [reply pkgs/top-level/all-packages.nix]: I was trying to figure out if that file is ordered; should have read the comment at the top instead :smile:

**Pull request merged**